library/src/main/java/com/owncloud/android/lib/resources/files/DownloadFileRemoteOperation.java

```
@@ -135,8 +138,55 @@ private int downloadFile(OwnCloudClient client, File targetFile) throws IOExcept
                     transferEncoding = "chunked".equals(transferEncodingHeader.getValue());
                 }
                 
-                if (transferred == totalToTransfer || transferEncoding) {  
-                    savedFile = true;
+                if (transferred == totalToTransfer || transferEncoding) {
```

If possible, could you apply the fail fast here?

---

What do you mean by that?

---

Instead of having nested if-else blocks, you can return early. However, I think the downloadFile() function is too complex for that. We can skip it if it’s too much work to refactor.

e.g.

Before:

```
public void checkAge(int age) {
    if (age >= 0) {
        if (age >= 18) {
            System.out.println("User is an adult.");
        } else {
            System.out.println("User is a minor.");
        }
    } else {
        System.out.println("Invalid age.");
    }
}
```

After:

```
public void checkAge(int age) {
    if (age < 0) {
        System.out.println("Invalid age.");
        return;  // Fail fast: stop if age is invalid
    }

    if (age >= 18) {
        System.out.println("User is an adult.");
        return;
    }
    
    System.out.println("User is a minor.");
}
```